ui-manchette/src/types.ts

```diff
@@ -1,73 +1,24 @@
 import { type CSSProperties } from 'react';
 
-export type PathProperties = {
-  operational_points?:
-    | {
-        extensions?: OperationalPointExtensions;
-        id: string;
-        part: OperationalPointPart;
-        /** Distance from the beginning of the path in mm */
-        position: number;
-      }[]
-    | null;
+export type Waypoint = {
+  id: string;
+  position: number; // in mm
+  name?: string;
+  secondaryCode?: string;
 };
 
-export type GeoJsonLineString = {
-  coordinates: GeoJsonLineStringValue;
-  type: 'LineString';
-};
-
-export type GeoJsonPointValue = number[];
-export type GeoJsonLineStringValue = GeoJsonPointValue[];
-
-export type OperationalPointExtensions = {
-  identifier?: {
-    name: string;
-    uic: number;
-  } | null;
-  sncf?: {
-    ch: string;
-    ch_long_label: string;
-    ch_short_label: string;
-    ci: number;
-    trigram: string;
-  } | null;
-};
-export type OperationalPointPart = {
-  extensions?: {
-    sncf?: {
-      kp: string;
-    } | null;
-  };
-  position: number;
-  track: string;
-};
-
-export type ArrayElement<ArrayType extends readonly unknown[] | null | undefined> =
-  ArrayType extends readonly (infer ElementType)[] ? ElementType : never;
-
-export type OperationalPointType = ArrayElement<PathProperties['operational_points'] | null>;
-
-export type StyledOperationalPointType = OperationalPointType & {
+export type EnrichedWaypoint = Waypoint & {
```

Comment:
Mandatory naming nitpicking: "enriched" is a bit vague. Would `InteractiveWaypoint` be better?